src/gui/properties/propertieswidget.cpp

```diff
@@ -335,6 +335,18 @@ void PropertiesWidget::loadTorrentInfos(BitTorrent::Torrent *const torrent)
         m_ui->labelCommentVal->setText(Utils::Misc::parseHtmlLinks(m_torrent->comment().toHtmlEscaped()));
 
         m_ui->labelCreatedByVal->setText(m_torrent->creator());
+
+        // Private
+        if (m_torrent->hasMetadata())
```

Comment:
>4. The bug that does not reset the private flag in the properties section of the GUI for torrents without metadata is still there.

This already exists in line 327 (at the moment of writing). https://github.com/qbittorrent/qBittorrent/pull/20951#issuecomment-2184100600 was referring to it.

/off topic
I can't help with API stuff.

Reply:
Thank you. Fixed it.